google/data_source_google_project.go

```diff
@@ -0,0 +1,44 @@
+package google
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func dataSourceGoogleProject() *schema.Resource {
```

Comment:
How do you feel about having the data source mirror the project resource? Take a look at the container cluster data source to see how we've handled that in other scenarios.